graph/appender/istio_details.go

```diff
@@ -36,17 +36,27 @@ func addRouteBadges(trafficMap graph.TrafficMap, namespace string, istioDetails
 func applyCircuitBreakers(trafficMap graph.TrafficMap, namespace string, istioDetails *kubernetes.IstioDetails) {
 NODES:
 	for _, n := range trafficMap {
-		version := n.Version
-		if version == graph.UnknownVersion {
-			version = ""
-		}
-
-		if destServices, ok := n.Metadata["destServices"]; ok {
-			for serviceName, _ := range destServices.(map[string]bool) {
-				for _, destinationRule := range istioDetails.DestinationRules {
-					if kubernetes.CheckDestinationRuleCircuitBreaker(destinationRule, namespace, serviceName, version) {
-						n.Metadata["hasCB"] = true
-						continue NODES
+		switch n.NodeType {
+		case graph.NodeTypeUnknown:
+			continue
+		case graph.NodeTypeService:
+			for _, destinationRule := range istioDetails.DestinationRules {
+				if kubernetes.CheckDestinationRuleCircuitBreaker(destinationRule, namespace, n.Service, "") {
+					n.Metadata["hasCB"] = true
+					continue NODES
+				}
+			}
+		default:
+			if n.Version == "" || n.Version == graph.UnknownVersion {
```

Comment:
Why skip if version is ""? If this is an app node, would it be possible for a CB to be on the node? i.e. doesn't CheckDestinationRuleCircuitBreaker accept a "" version?

Reply:
@jmazzitelli , I agree.  In fact I've re-thought and formalized the scenarios.  Please see [the related Jira comment](https://issues.jboss.org/browse/KIALI-1698?focusedCommentId=13643920&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13643920).  If you are OK with the scenarios you can then ensure the PR actualizes the design.